app/controllers/schools/users_controller.rb

```diff
@@ -30,8 +30,29 @@ def edit
     def create
       authorize! :manage_users, @school
       @user = User.new(user_params.merge(school: @school, created_by: current_user))
-      if @user.save
-        redirect_to school_users_path(@school)
+      # debugger
+      @user.valid?
+      if @user.role == 'school_admin' &&
+         @user.errors.where(:email).filter { |error| error.type != :taken }.empty? && !params.key?(:new_user_form)
+        existing_user = User.find_by(email: @user.email)
+        if existing_user&.role == 'group_admin'
+          skip_save = true
+          notice = "As a group admin for #{existing_user.school_group.name}, this user is already able to administer " \
+                     'this school'
+        elsif existing_user.present?
+          existing_user.add_cluster_school(@school)
+          existing_user.add_cluster_school(existing_user.school) unless existing_user.school.nil?
+          existing_user.role = :school_admin
+          @user = existing_user
+          notice = 'Added user as a school admin'
```

Comment:
do we want the alert contact create update logic here?

Reply:
I think it's fine. They can be subscribed to alerts later (either by themselves, or their school admin).